### ISSUE TITLE
Update Hapi API for breaking changes in 17

### ISSUE THREAD

**Issue by brabster** [bendrucker/hapi-require-https#24]
Plugins can't be used (without modification/third-party modules) with Hapi 17 unless they are updated for the new plugin API.

Breaking changes: https://github.com/hapijs/hapi/issues/3658

**Comment by bendrucker** [comment]
There’s a PR (#23) that’s really not there but tries to address this. Will accept a complete PR otherwise it might take me a week or two to look into this.

**Comment by brabster** [comment]
Never thought to check the PRs, just looked for an issue :smile: - I don't actually need this myself, wanted to let you know.

If I get a chance, I'll have a look!

**Comment by bendrucker** [comment]
Thanks! Seems like 17 is a huge change for plugins. Sounds like I need to do some research on clever ways to achieve backwards compat. May have time this week.

**Comment by brabster** [comment]
I came across https://www.npmjs.com/package/hapi-compat, not sure whether that works with hapi-require-https, but if so maybe an option to sidestep backwards compatibility.

The main reason I've upgraded to Hapi 17 is because one of the plugins I'm using has a feature I need that's only available after the plugin update and it's not backwards-compatible. Plugins like https://github.com/hapijs/vision also advertise that they won't work with <17. Might be OK to just update and see if anyone complains!
